keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/TimeWindowConstraintEvaluator.kt

```diff
@@ -0,0 +1,203 @@
+package com.netflix.spinnaker.keel.constraints
+
+import com.netflix.spinnaker.keel.api.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.TimeWindowConstraint
+import com.netflix.spinnaker.keel.constraints.ConstraintEvaluator.Companion.getConstraintForEnvironment
+import com.netflix.spinnaker.keel.exceptions.InvalidConstraintException
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService
+import org.springframework.stereotype.Component
+import java.text.ParsePosition
+import java.time.Clock
+import java.time.DayOfWeek
+import java.time.format.DateTimeFormatter
+import java.time.format.TextStyle
+import java.util.Locale
+import java.util.TimeZone
+import java.util.TimeZone.getTimeZone
+
+/**
+ * An environment promotion constraint to gate promotions to time windows.
+ * Multiple windows can be specified.
+ *
+ * Hours and days parameters support a mix of ranges and comma separated values.
+ *
+ * Days can be specified using full or short names according to the default JVM locale.
+ *
+ * A timezone can be set in the constraint and a system-wide default can by set via
+ * the dynamic property "default.time-zone" which defaults to UTC.
+ *
+ * Example env constraint:
+ *
+ * constraints:
+ *  - type: time-window
+ *    windows:
+ *      - days: Monday,Tuesday-Friday
```

Comment:
Up in a test file I saw [all lowercase for this value](https://github.com/spinnaker/keel/pull/586/files#diff-cb20a8e71cc1d6a198f43860b58d5696R149) — looks like in this evaluator we're coercing any capitalization into lowercase?

I don't have a problem with that necessarily, but might be good to shoot for a standardized way of doing examples.

Reply:
Or what if I add an additional window here that uses lower case? I want to show the flexibility in how days can be specified, but not sure how best to do that.

Reply:
yeah maybe? I dunno. I suspect all these constraints will need really thorough docs to be successful. It's a detail and probably not super important tbh.